src/devices/Sn74hc595/GpioControllerExtensions.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Linq;
+
+/// <summary>
+/// Extension methods for GpioController
+/// </summary>
+public static class GpioControllerExtensions
+{
+    /// <summary>
+    /// Opens a set of pins.
+    /// </summary>
+    public static void OpenPins(this GpioController controller, PinMode mode, params int[] pins)
+    {
+        foreach (var pin in pins)
+        {
+            controller.OpenPin(pin, mode);
+        }
+    }
+
+    /// <summary>
+    /// Opens a pin as PineMode.Output and writes value.
+    /// </summary>
+    public static void OpenPinAndWrite(this GpioController controller, int pin, PinValue value)
+    {
+        controller.OpenPin(pin, PinMode.Output);
+        controller.Write(pin, value);
+    }
+
+    /// <summary>
+    /// Writes a value to multiple pins.
+    /// </summary>
+    public static void WriteValueToPins(this GpioController controller, int value, params int[] pins)
```

Comment:
duplicate  `GpioController.Write(ReadOnlySpan<PinValuePair> pinValuePairs)` ?

Reply:
I assume you mean this: https://github.com/dotnet/iot/blob/master/src/System.Device.Gpio/System/Device/Gpio/GpioController.cs#L335

High-level, yes, it is a duplicate. Practically, it isn't. The referenced API has a lot of ceremony and duplication for writing a single value to multiple pins. If you write code with both APIs, you will immediately see how different they are.

Anyway, my top-level question is what to do with these extension methods, in general.

Reply:
> Put another way, the product already has higher-level helpers, and I find them too cumbersome to actually use in practice.

That is good feedback and we should probably take a deeper look there then.

It's fine if you want to have these helpers for your binding, but I would suggest a few things: a) have the helpers be internal as we don't want your binding to expose these if they are really just meant to be used internally, if you think they might be useful outside this binding, then let's move them to common and add a namespace. b) for this specific one, I would definitely call the Write that takes in the Span of PinValuePairs, the reason is that this specific method is optimized in some drivers, so if you instead use regular Write's in a loop, you are effectively loosing those perf gains that might be important for your binding.